### ISSUE TITLE
CONSIDER building two On This Page components for in-page anchor links

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#5125]
## Description

We are starting to use a pattern called "On This Page" that uses `<a>` tags to link to anchor IDs further down our pages. This technique works well, but has a few requirements:

* The IDs being anchored to need a `tabindex="-1"` attribute to work correctly with IE11
* Ideally the tab index is added dynamically with Javascript (possible snippet below)
* We will need to review all places `tabindex="-1"` is hard coded. Some of them are being used by React components to set focus on specific page sections, and should not be removed. Others like the related issue below should be added and cleaned up with dynamic JS.
* Will require a new Formation-react component, and a CMS Liquid template adjustment. Probably a platform helper function, too.

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/2429

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Possible Code Snippets

VSP and VSA feel this component should be a definition list. I've also found a decent way to check for IE and add/remove the tabindex that would need lifecycle methods, hooks, or custom event listeners depending on the context.

```html
<dl>
  <dt>On this page</dt>
  <dd><a href="#jump1">Jump 1</a></dd>
  <dd><a href="#jump2">Jump 2</a></dd>
 <dd><a href="#jump3">Jump 3</a></dd>
</dl>
```

```javascript
//  https://stackoverflow.com/a/49549464
const isIE = !!document.documentMode;

export const setTabIndex = el => {
  const target = document.getElementById(el);

  if (isIE) {
    target.setAttribute("tabindex", "-1");
  }
};

export const unsetTabIndex = el => {
  const target = document.getElementById(el);

  if (isIE) {
    target.removeAttribute("tabindex");
  }
};
```

**Comment by 1Copenut** [comment]
Issue moved to [department-of-veterans-affairs/vets-design-system-documentation #245](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/245) via [**ZenHub**](https://www.zenhub.com/)
